### PR TITLE
Reduce number of calls for user

### PR DESCRIPTION
**Pull request by 8r2y5**
I want to merge this change because this will reduce number of calls when retrieving user 

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

**Review by maarcingebala — COMMENTED** [review]
We could add unit tests to catch the plugin_manager_promise issue when requestor was `None`.

**Review by SzymJ — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/plugins/dataloaders.py [hunk 1]
```diff
@@ -46,6 +47,12 @@ def batch_load(self, keys):
 def plugin_manager_promise(context: SaleorContext, app) -> Promise[PluginsManager]:
     user = context.user
     requestor = app or user
+
+    if isinstance(requestor, SimpleLazyObject):
+        if requestor._wrapped is empty:  # type: ignore[attr-defined]
```
Comment (wcislo-saleor) [inline saleor/graphql/plugins/dataloaders.py:52]: Reuse this function?

https://github.com/saleor/saleor/blob/2e261eae739660da427e5f40e1efc6e19639dc1b/saleor/core/utils/lazyobjects.py#L35-L39

**Review by wcislo-saleor — COMMENTED** [review]
Do we know how to reproduce a request where Saleor makes redundant calls to plugin's `authenticate_user`?

**Comment by 8r2y5** [comment]
Fix was done in https://github.com/saleor/saleor/pull/14299
